### DISCUSSION TITLE
api routes not loading?

### DISCUSSION THREAD

**Discussion by christaggart** [vercel/vercel#7300] [Help]
When running `vercel dev` within a simple vue3/vite app hitting `http://localhost:3000/api/hello`  simply loads the vue app, instead of executing the node function.

One thing I noticed was that when running with explicit port specification, vite tries to run on port 3000 after the node function already starts to run on port 3000, so vite automatically starts running on port 3001. I can then try `vc dev --debug --listen 4000` at which point vite starts on port 3000, and the service proxied via port 4000 eventually times out with the following error:
```
 [debug] [2022-01-09T19:03:32.462Z] Closing file watcher
Error! Detecting port 64599 timed out after 300000ms
> [debug] [2022-01-09T19:03:32.530Z] Error: Detecting port 64599 timed out after 300000ms``` 

I assume this is supposed to be the proxy itself or the node function? Is there a way to debug this thing that runs the node function?
```

_package.json snip_
```
"scripts": {
    "dev": "vite --port $PORT",
    "build": "vite build",
    "serve": "vite preview"
  }, 
 ```
[debug] [2022-01-09T18:57:12.190Z] No Runtimes need to be installed
> [debug] [2022-01-09T18:57:12.193Z] Requiring runtime: "@vercel/node"
> [debug] [2022-01-09T18:57:12.514Z] Adding build match for "api/hello" with "@vercel/node@latest"

  Dev server running at:
  > Local:    http://localhost:3000/
  > Network:  http://192.168.1.30:3000/

You can see it finds the build match for api/hello, but hitting that just loads the regular vite app.

**Comment by andrewgar0122** [comment]
I am also experiencing bugging and [lagging ](https://techlarapoint.com/project-qt-mod-apk/)issues which are nerve-racking.
